### PR TITLE
technititum-dns-server: 13.2 -> 13.2.2

### PR DESCRIPTION
**Pull request by FabianRig**
Update Technitium DNS Server and its library to 13.2.2. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/te/technitium-dns-server-library/package.nix [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
-  version = "dns-server-v13.2";
+  version = "dns-server-v13.2.2";
 
   src = fetchFromGitHub {
     owner = "TechnitiumSoftware";
     repo = "TechnitiumLibrary";
     rev = "refs/tags/${version}";
-    hash = "sha256-stfxYe0flE1daPuXw/GAgY52ZD7pkqnBIBvmSVPWWjI=";
+    hash = "sha256-KJNz4jhCKySYrYkT3BxqBh9QkeDV0XGHlRg9K0jLvBI=";
     name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/by-name/te/technitium-dns-server-library/package.nix:17]:
```suggestion
```

File: pkgs/by-name/te/technitium-dns-server/package.nix [hunk 2]
```diff
@@ -9,13 +9,13 @@
 }:
 buildDotnetModule rec {
   pname = "technitium-dns-server";
-  version = "13.2";
+  version = "13.2.2";
 
   src = fetchFromGitHub {
     owner = "TechnitiumSoftware";
     repo = "DnsServer";
     rev = "refs/tags/v${version}";
```
Comment (SuperSandro2000) [inline pkgs/by-name/te/technitium-dns-server/package.nix:17]:
```suggestion
    tag = "v${version}";
```

File: pkgs/by-name/te/technitium-dns-server-library/package.nix [hunk 3]
```diff
@@ -7,13 +7,13 @@
 }:
 buildDotnetModule rec {
   pname = "technitium-dns-server-library";
-  version = "dns-server-v13.2";
+  version = "dns-server-v13.2.2";
 
   src = fetchFromGitHub {
     owner = "TechnitiumSoftware";
     repo = "TechnitiumLibrary";
     rev = "refs/tags/${version}";
```
Comment (SuperSandro2000) [inline pkgs/by-name/te/technitium-dns-server-library/package.nix:15]:
```suggestion
    tag = version;
```

File: pkgs/by-name/te/technitium-dns-server-library/package.nix [hunk 4]
```diff
@@ -7,13 +7,13 @@
 }:
 buildDotnetModule rec {
   pname = "technitium-dns-server-library";
-  version = "dns-server-v13.2";
+  version = "dns-server-v13.2.2";
```
Comment (SuperSandro2000) [inline pkgs/by-name/te/technitium-dns-server-library/package.nix:10]:
```suggestion
  version = "dns-server-v${technitium-dns-server.version}";
```

**Comment by FabianRig** [comment]
@SuperSandro2000 Thank you for your review! I've applied all changes, but now the build is failing. Perhaps because of
```
inherit (technitium-dns-server) version;
```
? 

Any suggestions what to edit/add, please?

**Comment by timhae** [comment]
[Yes](https://github.com/NixOS/nixpkgs/actions/runs/12321724094/job/34393742348?pr=363529#step:5:443), you need to pass in the package at the top so callPackage picks it up but then you might have an infinite recursion.. Which would speak for merging the packages again.. And then create an update script that updates both

**Comment by FabianRig** [comment]
OK. I'd like to wait if @SuperSandro2000 has an idea how to solve this elegantly. If not, I'd close this PR and probably go back to joining the two packages again. :-(

**Comment by timhae** [comment]
I asked in matrix, there doesn't seem to be a way. You can join the packages again and I will try to create an update script that covers both

**Comment by FabianRig** [comment]
@timhae: Could you please check https://github.com/NixOS/nixpkgs/pull/365428 and review it?

**Pull request closed without merging**